db/migrate/20220516111152_add_team_level_permissions.rb

```diff
@@ -1,6 +1,18 @@
 # frozen_string_literal: true
 
 class AddTeamLevelPermissions < ActiveRecord::Migration[6.1]
+  class UserAssignment < ApplicationRecord
```

Comment:
Why it is required to define this model here again?

Reply:
Because we later added team to UserAssignments, so this migration will fail with current model, as the team_id validation will fail.

Reply:
This validation?

  validates :user, uniqueness: { scope: %i(assignable team_id) }

Reply:
This also, but `belongs_to :team` also generates a presence validator for team.

Reply:
We can make uniqueness validation conditional, and always set team for belongs_to

Reply:
We can't, because at the point of this migration, UserAssignment doesn't have team_id yet. It was added in a later migration (20220621153016_add_team_reference_to_user_assignments.rb).